### PR TITLE
use flush and reorder action to get proper tlm during deletion/anonym…

### PR DESCRIPTION
**Pull request by inkhey**
…ization or user

related to #4746 

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [x] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

### CODE REVIEW

File: backend/tracim_backend/__init__.py [hunk 1]
```diff
@@ -254,7 +254,14 @@ def web(global_config: OrderedDict, **local_settings) -> Router:
     context = TracimPyramidContext(
         configurator=configurator, default_error_builder=ErrorSchema(), debug=app_config.DEBUG
     )
-    hapic.set_context(context)
+    # HACK - G.M - 2021-07-01 - Force reset of context if already set.
+    # This case happened in some test now for unclear reasons.
```
Comment (raphj) [inline backend/tracim_backend/__init__.py:258]: Do we plan to / is there a way get rid of this hack? If so, maybe write an issue, if not, you can simply resolve this conversation.

**Review by raphj — APPROVED** [review]

File: backend/tracim_backend/__init__.py (hunk 1 above)
Comment (inkhey) [reply backend/tracim_backend/__init__.py:258]: I suspect it's a case that happened only in command test as we go multiple time in this code. but i'm not really sure.

File: backend/tracim_backend/lib/core/event.py [hunk 2]
```diff
@@ -836,16 +837,26 @@ def _has_just_been_undeleted(self, obj: Union[User, Workspace, ContentRevisionRO
...
+def get_event_user_id(session: TracimSession, event: Event) -> typing.Optional[int]:
+    try:
+        user_id = session.query(User.user_id).filter(User.user_id == event.user["user_id"]).one()
+        return user_id[0]
+    except AttributeError:
+        # no user in event
+        return None
+    except NoResultFound:
+        # user doesn't exist anymore
+        return None
```
Comment (grignards) [inline backend/tracim_backend/lib/core/event.py:849]:
```suggestion
    except (AttributeError, NoResultFound):
        # no user in event or user doesn't exist anymore
        return None

**Review by grignards — APPROVED** [review]

**Comment by gbonaspetti** [comment]
Tested on branch

**Pull request merged**